File: DemoApp/DemoApp/ADVSInterstitialViewController.m
//
//  ADVSInterstitialViewController.m
//  DemoApp
//
//  Created by M.T.Burn on 2014/10/17.
//  Copyright (c) 2014年 MTBurn. All rights reserved.
//

#import "ADVSInterstitialViewController.h"
#import <AppDavis/ADVSInterstitialAdLoader.h>
#import "ADVSAppDelegate.h"

@interface ADVSInterstitialViewController () <ADVSInterstitialAdLoaderDelegate>
@end

@implementation ADVSInterstitialViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    // Removed comment out if you want to stop skipping interstitial ads.
    // (Setting) adspot_id: OTM0OjQ1Nw and OTM0OjQ2MQ are set that skip count is 2 times until first display and 1 times after that.
    //  See http://gyazo.com/e279ae315a31ff32a8505528dc940782 (in japanese)
    NSString *adspot_id = [self getAdSpotId];
    if (0 < [adspot_id length]) {
        [[ADVSInterstitialAdLoader sharedInstance] setAdSpotId:adspot_id];
    }
    
    [ADVSInterstitialAdLoader sharedInstance].delegate = self;
    [[ADVSInterstitialAdLoader sharedInstance] loadRequest];
}

- (NSString*)getAdSpotId
{
    ADVSAppDelegate *delegate = (ADVSAppDelegate *) [[UIApplication sharedApplication] delegate];
    return [delegate adspotIdDict][@"interstitial"];
}

#pragma mark - ADVSInterstitialAdLoaderDelegate

- (void)interstitialAdLoaderDidStartLoadingAd:(ADVSInterstitialAdLoader *)interstitialAdLoader
{
    NSLog(@"interstitialAdLoaderDidStartLoadingAd");
}

- (void)interstitialAdLoaderDidFinishLoadingAd:(ADVSInterstitialAdLoader *)interstitialAdLoader
{
    NSLog(@"interstitialAdLoaderDidFinishLoadingAd");
}

- (void)interstitialAdLoaderDidFinishLoadingAdView:(ADVSInterstitialAdLoader *)interstitialAdLoader
{
    NSLog(@"interstitialAdLoaderDidFinishLoadingAdView");
    
    [[ADVSInterstitialAdLoader sharedInstance] displayAd];
    
    // Removed a comment out if you want to try setting red to background color.
    // [[ADVSInterstitialAdLoader sharedInstance] displayAd:[UIColor redColor]];
}

- (void)interstitialAdLoaderDidSkipLoadingAd:(ADVSInterstitialAdLoader *)interstitialAdLoader
{
    NSLog(@"interstitialAdLoaderDidSkipLoadingAd");
}

- (void)interstitialAdLoaderDidClickIntersititialAdView:(ADVSInterstitialAdLoader *)interstitialAdLoader
{
    NSLog(@"interstitialAdLoaderDidClickIntersititialAdView:");
}

- (void)interstitialAdLoaderDidCloseIntersititialAdView:(ADVSInterstitialAdLoader *)interstitialAdLoader
{
    NSLog(@"interstitialAdLoaderDidCloseIntersititialAdView:");
}

- (void)interstitialAdLoader:(ADVSInterstitialAdLoader *)interstitialAdLoader didFailToLoadAdWithError:(NSError *)error
{
    NSLog(@"interstitialAdLoader:didFailToLoadAdWithError:%@", error);
}

- (void)interstitialAdLoader:(ADVSInterstitialAdLoader *)interstitialAdLoader didFailToLoadAdViewWithError:(NSError *)error
{
    NSLog(@"interstitialAdLoader:didFailToLoadAdViewWithError:%@", error);
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
}

@end

File: DemoAppSwift/DemoAppSwift/ADVSInterstitialViewController.swift
//
//  ADVSInterstitialViewController.swift
//  DemoAppSwift
//
//  Created by M.T.Burn on 2015/09/11.
//  Copyright © 2015年 M.T.Burn. All rights reserved.
//

import UIKit

class ADVSInterstitialViewController: UIViewController, ADVSInterstitialAdLoaderDelegate {
    
    let adSpotId = "OTM0OjQ2MQ"
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        // Load HIKE advertisements
        ADVSInterstitialAdLoader.sharedInstance().delegate = self;
        ADVSInterstitialAdLoader.sharedInstance().loadRequest()
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
    }
    
    // MARK: - HIKE delegate methods
    func interstitialAdLoaderDidStartLoadingAd(interstitialAdLoader: ADVSInterstitialAdLoader!) {
        NSLog("interstitialAdLoaderDidStartLoadingAd")
    }
    
    func interstitialAdLoaderDidFinishLoadingAd(interstitialAdLoader: ADVSInterstitialAdLoader!) {
        NSLog("interstitialAdLoaderDidFinishLoadingAd")
    }
    
    func interstitialAdLoaderDidFinishLoadingAdView(interstitialAdLoader: ADVSInterstitialAdLoader!) {
        NSLog("interstitialAdLoaderDidFinishLoadingAdView")
        
        ADVSInterstitialAdLoader.sharedInstance().displayAd()
    }
    
    func interstitialAdLoaderDidSkipLoadingAd(interstitialAdLoader: ADVSInterstitialAdLoader!) {
        NSLog("interstitialAdLoaderDidSkipLoadingAd")
    }
    
    func interstitialAdLoaderDidClickIntersititialAdView(interstitialAdLoader: ADVSInterstitialAdLoader!) {
        NSLog("interstitialAdLoaderDidClickIntersititialAdView")
    }
    
    func interstitialAdLoaderDidCloseIntersititialAdView(interstitialAdLoader: ADVSInterstitialAdLoader!) {
        NSLog("interstitialAdLoaderDidCloseIntersititialAdView")
    }
    
    func interstitialAdLoader(interstitialAdLoader: ADVSInterstitialAdLoader!, didFailToLoadAdWithError error: NSError!) {
        NSLog("interstitialAdLoader:didFailToLoadAdWithError:%@", error)
    }
    
    func interstitialAdLoader(interstitialAdLoader: ADVSInterstitialAdLoader!, didFailToLoadAdViewWithError error: NSError!) {
        NSLog("interstitialAdLoader:didFailToLoadAdViewWithError:%@", error)
    }
}
